data/hai/hai missions.txt

```diff
@@ -109,6 +109,16 @@ mission "Discovered Wormhole Alpha"
 		set "wormhole alpha found"
 		fail
 
+# Allows setting of the "hai discovered" condition even if first contact has already occurred
+mission "Hai Discovered Patch"
+	landing
+	invisible
+	source
+		government "Hai" "Hai (Unfettered)"
+	on offer
+		set "discovered hai"
+		fail
+
```

Comment:
With the suggested change to "First Contact: Wanderer" this is no longer needed, but I do think you should update the "Discovered Hai Space" mission further up in the file to also offer on landing on an Unfettered world.